api/middleware/globalThrottler.go

```diff
@@ -85,6 +85,11 @@ func (gt *globalThrottler) printDebugInfo() {
 	log.Debug(fmt.Sprintf("API engine stuck: \n%s", strings.Join(infoLines, "\n")))
 }
 
+// Type returns the type of the middleware
+func (gt *globalThrottler) Type() string {
```

Comment:
add tests for all implementations in which you will have nil pointer receivers as to properly test the code does not panic

Reply:
removed the method